File: Backend/src/main/java/com/workout/tracker/entity/Workout.java
package com.workout.tracker.entity;

import com.workout.tracker.security.entity.UserEntity;
import jakarta.persistence.*;
import lombok.Data;

import java.util.List;

@Data
@Entity
public class Workout {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private Integer icon;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private UserEntity user;

    @OneToMany(mappedBy = "workout", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<WorkoutPerDay> workoutPerDays;
}

File: Frontend/src/app/class/workout.ts
import { WorkoutPerDay } from './workoutPerDay';

export class Workout {
  public id?: number;
  public name: string;
  public icon: number;
  public workoutPerDays: WorkoutPerDay[];

  public constructor(
    name: string,
    workoutPerDays: WorkoutPerDay[] = [],
    icon: number = 0,
    id?: number
  ) {
    this.id = id;
    this.name = name;
    this.icon = icon;
    this.workoutPerDays = workoutPerDays || [];
  }
}
